File: 19-06-2025_Case-specific_Sorting_of_Strings.cpp
class Solution {
public:
    string caseSort(string s) {
        string lower, upper;
        for (char c : s) {
            if (islower(c)) lower += c;
            else upper += c;
        }
        
        sort(lower.begin(), lower.end());
        sort(upper.begin(), upper.end());

        string result = "";
        int i = 0, j = 0;
        for (char c : s) {
            if (islower(c)) {
                result += lower[i++];
            } else {
                result += upper[j++];
            }
        }
        return result;
    }
};

File: 19-06-2025_Case-specific_Sorting_of_Strings.java
class Solution {
    public String caseSort(String s) {
        List<Character> lower = new ArrayList<>();
        List<Character> upper = new ArrayList<>();
        
        for (char c : s.toCharArray()) {
            if (Character.isLowerCase(c)) lower.add(c);
            else upper.add(c);
        }

        Collections.sort(lower);
        Collections.sort(upper);

        StringBuilder result = new StringBuilder();
        int i = 0, j = 0;

        for (char c : s.toCharArray()) {
            if (Character.isLowerCase(c)) {
                result.append(lower.get(i++));
            } else {
                result.append(upper.get(j++));
            }
        }

        return result.toString();
    }
}
